src/components/QuranReader/SidebarNavigation/SurahList.tsx

```diff
@@ -0,0 +1,65 @@
+import { useState } from 'react';
+
+import classNames from 'classnames';
+import Fuse from 'fuse.js';
+import useTranslation from 'next-translate/useTranslation';
+
+import styles from './SidebarNavigation.module.scss';
+
+import Link from 'src/components/dls/Link/Link';
+import useChapterIdsByUrlPath from 'src/hooks/useChapterId';
+import { getAllChaptersData } from 'src/utils/chapter';
+import Chapter from 'types/Chapter';
+
+const filterSurah = (surah, searchQuery: string) => {
+  const fuse = new Fuse(surah, {
+    threshold: 0.3,
+    keys: ['id', 'transliteratedName'],
+  });
+
+  const filteredSurah = fuse.search(searchQuery).map(({ item }) => item);
+  return filteredSurah as Chapter[];
+};
+
+const SurahList = () => {
+  const chapterIds = useChapterIdsByUrlPath();
+  const currentChapterId = chapterIds[0];
+  const { t } = useTranslation('common');
+
+  const chaptersData = getAllChaptersData();
```

Comment:
Locale/lang needs to be passed to `getAllChaptersData` so that the Surah names will be localized. You can get the locale value from `const { t, lang } = useTranslation('common');`